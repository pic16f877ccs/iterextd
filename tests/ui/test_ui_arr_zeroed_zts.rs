use iterextd::IterExtd;

fn main() {
    let mut arr = [10u8, 11, 22, 33, 44, 55, 66, 77, 88, 99, 111, 222];
    let arr_sl: [&[u8]; 3] = [&arr[0..3], &arr[3..6], &arr[6..9]];
    let iter = arr_sl.iter().arr_chunks::<2>().array_copied();
    let _: (usize, [_; 2]) = iter.collect_arr_zeroed();
}
