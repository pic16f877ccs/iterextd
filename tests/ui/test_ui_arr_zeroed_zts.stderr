error[E0277]: the trait bound `&[u8]: iterextd::iterator::AllowZero` is not satisfied
 --> tests/ui/test_ui_arr_zeroed_zts.rs:7:35
  |
7 |     let _: (usize, [_; 2]) = iter.collect_arr_zeroed();
  |                                   ^^^^^^^^^^^^^^^^^^ the trait `iterextd::iterator::AllowZero` is not implemented for `&[u8]`
  |
  = help: the trait `iterextd::iterator::AllowZero` is implemented for `[T; N]`
  = note: required for `[&[u8]; 2]` to implement `iterextd::iterator::AllowZero`
note: required by a bound in `collect_arr_zeroed`
 --> src/iterator.rs
  |
  |     fn collect_arr_zeroed<const N: usize>(self) -> (usize, [Self::Item; N])
  |        ------------------ required by a bound in this associated function
...
  |         Self::Item: AllowZero,
  |                     ^^^^^^^^^ required by this bound in `IterExtd::collect_arr_zeroed`
